wordpress-mobile/WordPress-iOS
# Signup Epilogue: update table and cells for new design

Ref #13939 
Related WPAuth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/252 (to add the secure text toggle button to the Password field)

This updates the Signup Epilogue table and Account Details cells to match the new design. 

No functionality has changed, with the exception of the secure text entry toggle on the Password field. Tapping it will hide/show the text entered in the Password field.

---
Known issues:

Password field:
- There is a leading margin that makes the field inset by 8 points.
- The secure text toggle icon color doesn't match the design.

Username field:
- The disclosure icon is inset 4 points.

These use a custom field type defined in WPAuth, and is used in several other auth views. I'll look at these separately since I'll have to figure out how to update this view without affecting the others.

---
To test:
- Signup. (See below for Auth hack to avoid actually signing up.)
- On the Epilogue, verify everything above the `Continue` button looks like the design on #13939. 

| Before | After |
|--------|-------|
| ![before_light](https://user-images.githubusercontent.com/1816888/80052086-0aaa7600-84d7-11ea-898a-c5c1e79b3ae4.png) | ![after_light](https://user-images.githubusercontent.com/1816888/80137991-61f52880-8561-11ea-9d30-14d9d3dd2487.png) |
| ![before_dark](https://user-images.githubusercontent.com/1816888/80052102-15650b00-84d7-11ea-8f5e-2433b042809a.png) | ![after_dark](https://user-images.githubusercontent.com/1816888/80138024-6b7e9080-8561-11ea-8b50-430b13576699.png) |

---
To show the Signup Epilogue without actually signing up:

- Check out the accompanying `WordPressAuthenticator` branch locally. (`issue/13939-secure_text_entry_toggle`)
- In WPiOS `Podfile`, point `WordPressAuthenticator` to your local Auth.
- In `LoginViewController`, change `showLoginEpilogue` to show the signup epilogue:

```
func showLoginEpilogue(for credentials: AuthenticatorCredentials) {
    guard let navigationController = navigationController else {
        fatalError()
    }
    
    showSignupEpilogue(for: credentials)
    
//        authenticationDelegate.presentLoginEpilogue(in: navigationController, for: credentials) { [weak self] in
//            self?.dismissBlock?(false)
//        }
}
```

- _Login_ with an existing account, and the _signup_ epilogue will be displayed.

---
PR submission checklist:
